src/parser/TextSearchQuery.cpp

```diff
@@ -0,0 +1,202 @@
+    predStringTextSearch(subject.getVariable(), object.getVariable());
+  } else if (predString == "contains-word") {
+    throwSubjectVariableException("contains-word", subject);
+    throwContainsWordOrEntity(subject);
+    throwObjectLiteralException("contains-word", object);
+    predStringContainsWord(subject.getVariable(), object.getLiteral());
+  } else if (predString == "contains-entity") {
+    throwSubjectVariableException("contains-entity", subject);
+    throwContainsWordOrEntity(subject);
+    predStringContainsEntity(subject.getVariable(), object);
+  } else if (predString == "bind-match") {
+    throwSubjectAndObjectVariableException("bind-match", subject, object);
+    predStringBindMatch(subject.getVariable(), object.getVariable());
+  } else if (predString == "bind-score") {
+    throwSubjectAndObjectVariableException("bind-score", subject, object);
+    predStringBindScore(subject.getVariable(), object.getVariable());
+  }
+}
+
+// ____________________________________________________________________________
+std::vector<std::variant<TextIndexScanForWordConfiguration,
+                         TextIndexScanForEntityConfiguration>>
+TextSearchQuery::toConfigs(const QueryExecutionContext* qec) const {
+  std::vector<std::variant<TextIndexScanForWordConfiguration,
+                           TextIndexScanForEntityConfiguration>>
+      output;
+  // First pass to get all word searches
+  ad_utility::HashMap<Variable, vector<string>> potentialTermsForCvar;
+  for (const auto& [var, conf] : configVarToConfigs_) {
+    if (!conf.isWordSearch_.has_value()) {
+      throw TextSearchException(
+          "Text search service needs configs with exactly one occurrence of "
+          "either <contains-word> or "
+          "<contains-entity>.");
+    }
+    if (!conf.textVar_.has_value()) {
+      AD_FAIL();
+    }
+    if (conf.isWordSearch_.value()) {
+      potentialTermsForCvar[conf.textVar_.value()].push_back(
+          conf.word_.value());
+    }
+  }
+  // Second pass to create all configs
+  for (const auto& [var, conf] : configVarToConfigs_) {
+    if (conf.isWordSearch_.value()) {
+      output.emplace_back(TextIndexScanForWordConfiguration{
+          conf.textVar_.value(), conf.word_.value(), conf.varToBindMatch_,
+          conf.varToBindScore_});
+    } else {
+      auto it = potentialTermsForCvar.find(conf.textVar_.value());
+      if (it == potentialTermsForCvar.end()) {
+        throw TextSearchException(absl::StrCat(
+            "Entity search has to happen on a text Variable that is also "
+            "contained in a word search. Text Variable: ",
+            conf.textVar_.value().name(),
+            " is not contained in a word search."));
+      }
+      output.emplace_back(TextIndexScanForEntityConfiguration{
+          conf.textVar_.value(), conf.entity_.value(),
+          it->second[qec->getIndex().getIndexOfBestSuitedElTerm(it->second)],
+          conf.varToBindScore_});
+    }
+  }
+  return output;
```

Comment:
The old `ql:contains-word` has special semantics for strings like "alpha beta blubb*" 
wich actually splits into three words "alpha" , "beta" and the prefix "blubb*".
I still have to think what we shall do in this case (Maybe emit a warning, that becomes part of the created config (then you can propagate it in the query planner)